testament/categories.nim

```diff
@@ -424,11 +424,12 @@ proc listPackages(packageFilter: string): seq[NimblePackage] =
     # at least should be a regex; a substring match makes no sense.
     result = pkgs.filterIt(packageFilter in it.name)
   else:
-    let pkgs1 = pkgs.filterIt(it.allowFailure)
-    let pkgs2 = pkgs.filterIt(not it.allowFailure)
-    if testamentData0.batchArg == "allowed_failures":
-      result = pkgs1
+    if testamentData0.testamentNumBatch == 0:
+      result = pkgs
+    elif testamentData0.batchArg == "allowed_failures":
+      result = pkgs.filterIt(it.allowFailure)
```

Comment:
```suggestion
    if testamentData0.batchArg == "allowed_failures":
      return pkgs.filterIt(it.allowFailure)
    doAssert testamentData0.batchArg.len == 0, testamentData0.batchArg
    if testamentData0.testamentNumBatch == 0:
      result = pkgs
```

otherwise it won't work (as seen with CI failure for `allowed_failures` batch)

Reply:
Thanks!